fbw-a32nx/src/systems/fmgc/src/guidance/vnav/descent/TodGuidance.ts

```diff
@@ -44,61 +47,68 @@ export class TodGuidance {
 
   update(deltaTime: number) {
     this.updateTdReached(deltaTime);
-    this.updateTdPause(deltaTime);
+    if (NXDataStore.get('PAUSE_AT_TOD', 'DISABLED') === 'ENABLED') {
+      this.updateTdPause(deltaTime);
+    }
   }
 
   updateTdPause(deltaTime: number) {
-    if (this.cooldown <= 0 && NXDataStore.get('PAUSE_AT_TOD', 'DISABLED') === 'ENABLED') {
-      // Only watching if T/D pause untriggered + between flight phase CLB and CRZ
+    // Only armed if all conditions met
+    this.tdArmed =
+      this.cooldown <= 0 &&
+      !this.tdPaused &&
+      this.observer.get().flightPhase >= FmgcFlightPhase.Climb &&
+      this.observer.get().flightPhase <= FmgcFlightPhase.Cruise &&
+      Simplane.getAutoPilotAirspeedManaged();
+
+    if (SimVar.GetSimVarValue('L:A32NX_PAUSE_AT_TOD_ARMED', 'bool') !== this.tdArmed)
+      SimVar.SetSimVarValue('L:A32NX_PAUSE_AT_TOD_ARMED', 'bool', this.tdArmed);
+
+    if (this.tdArmed) {
+      // Check T/D pause first
       if (
-        !this.tdPaused &&
-        this.observer.get().flightPhase >= FmgcFlightPhase.Climb &&
-        this.observer.get().flightPhase <= FmgcFlightPhase.Cruise &&
-        Simplane.getAutoPilotAirspeedManaged()
+        (this.aircraftToDescentProfileRelation.distanceToTopOfDescent() ?? Number.POSITIVE_INFINITY) <
+        parseFloat(NXDataStore.get('PAUSE_AT_TOD_DISTANCE', '10'))
```

Comment:
Would be good to use `NXDataStore.getAndSubscribe(...)` (e.g. in the constructor) and set the values in a member variable and use that in the loop. I think NXDataStore.get is relatively slow and the value does not change often

Reply:
> Would be good to use `NXDataStore.getAndSubscribe(...)` (e.g. in the constructor) and set the values in a member variable and use that in the loop. I think NXDataStore.get is relatively slow and the value does not change often

Unfortunately I tried this once and it introduced a bug that somehow broke the entire functionality. I could try again at some point but as this part has essentially been in the codebase for months without causing significant observed performance degradation, I won't push this for now.

Probably would be better to write a new utility for this that could be abstract enough to be used everywhere, then replace all such NXDataStore operations in all js/ts.

Reply:
The change in this PR merely rearranges the logic slightly. Firstly to decouple the logic gating of the backoff timer from the NXDataStore check, such that this function ``updateTdPause`` is not called if ``PAUSE_AT_TOD`` is disabled. 

This enables this ``tdArmed`` state to be set (regardless of the status of the backoff timer)